pkg/user/handler.go

```diff
@@ -317,3 +318,50 @@ func (h Handler) Delete(c *gin.Context) {
 
 	c.Status(http.StatusAccepted)
 }
+
+type UpdateUserRequest struct {
+	Email    string `json:"email" binding:"omitempty,email"`
+	Password string `json:"password" binding:"omitempty,gte=16,lte=128"`
+}
+
+// Update user
+func (h *Handler) Update(c *gin.Context) {
+	// swagger:route PUT /users/{id} updateUser
+	//
+	// Update user
+	//
+	// Update user's email and/or password
+	//
+	// security:
+	//   oauth2:
+	//
+	// responses:
+	//   200: User
+	//   401: Error
+	//   403: Error
+	//   404: Error
+	//   415: Error
+	id, ok := handler.GetPathParameter(c, "id")
+	if !ok {
+		return
+	}
+
+	var request UpdateUserRequest
+	if err := handler.DataBinder(c, &request); err != nil {
+		_ = c.Error(err)
```

Comment:
@tonsV2 @radnov there seems to be a pattern of handler methods that process user input. When the processing fails we should stop processing the request setting the error into the Gin context. For example `GetPathParameter` (which you adapted), `GetUserFromContext` see https://github.com/dhis2-sre/im-manager/pull/257#discussion_r1223792766 and `DataBinder` as seen here.

We could adapt such methods in a way similar to `GetPathParameter` so here it would be

```go
if !handler.DataBinder(c, &request) {
  return
}
```

with `DataBinder` setting the error into the Gin context returning false on error. The downside of such signatures is that with returning `error` its clearer that we should stop processing than it with with a method returning a `bool`. Ignoring a returned `error` is also caught by linting while `bool` is not. Whatever we choose we should be consistent. Thinking through this I actually favor returning `error` 🤔

Reply:
I'm also in favour of returning an `error` instead of `bool`. 👍  I suppose adapting the `GetUserFromContext` and `DataBinder` methods to be as `GetPathParameter` should be done in a seaprate PR, though?

Reply:
If we are in favor of returning an `error` than we don't need to change `DataBinder` and `GetUserFromContext`. The intention of `GetPathParameter` was to remove repetition, not have to do the type conversion to uint and set the correct HTTP status code https://github.com/dhis2-sre/im-manager/blob/f3e6a2dc4d4b05c45592c6c782bde8656ac57fe1/internal/handler/pathParameter.go#L15. Due to the latter there is no need to return an error as its already in the Gin context. Lets leave this for now.